lib/cocoapods/sources_manager.rb

```diff
@@ -208,22 +255,19 @@ def update(source_name = nil, show_output = false)
           sources =  git_sources
         end
 
+        changed_spec_paths = {}
         sources.each do |source|
           UI.section "Updating spec repo `#{source.name}`" do
-            Dir.chdir(source.repo) do
-              begin
-                output = git! %w(pull --ff-only)
-                UI.puts output if show_output && !config.verbose?
-              rescue Informative
-                UI.warn 'CocoaPods was not able to update the ' \
-                  "`#{source.name}` repo. If this is an unexpected issue " \
-                  'and persists you can inspect it running ' \
-                  '`pod repo update --verbose`'
-              end
-            end
+            changed_spec_paths[source] = source.update(show_output && !config.verbose?)
             check_version_information(source.repo)
           end
         end
+        # Perform search index update operation as a subprocess.
+        Process.fork do
```

Comment:
can we extract this bit into a new method?
`update_search_index_if_needed_in_background` or something?